django/contrib/auth/views.py

```diff
@@ -442,11 +445,21 @@ class PasswordResetConfirmView(PasswordContextMixin, FormView):
     @method_decorator(never_cache)
     def dispatch(self, *args, **kwargs):
         assert 'uidb64' in kwargs and 'token' in kwargs
+        uidb64 = kwargs['uidb64']
+        token = kwargs['token']
 
         self.validlink = False
-        self.user = self.get_user(kwargs['uidb64'])
+        self.user = self.get_user(uidb64)
+
+        if token == INTERNAL_RESET_TOKEN:
+            token = self.request.session[INTERNAL_RESET_TOKEN]
+        else:
+            self.request.session[INTERNAL_RESET_TOKEN] = token
+            return HttpResponseRedirect(reverse('password_reset_confirm',
```

Comment:
You should fetch the arguments and url name from `request.resolver_match` here to ensure that we redirect to the same view, if someone hooks up `password_reset_confirm` with a different name you'd get an error here.